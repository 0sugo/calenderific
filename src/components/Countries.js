import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { useState } from 'react';
import { fetchCountryDetails } from '../redux/InfoCountry/InfoCountrySlice';
import { RightIcon } from './Icons';
import Navbar from './Navbar';

const Countries = () => {
  const { AllHolidays } = useSelector((store) => store.allHolidays);
  const dispatch = useDispatch();
  const [search, setSearch] = useState('');
  const handleDetailsFetch = (e) => {
    const { id } = e.target;
    const url = `https://date.nager.at/api/v3/publicholidays/2023/${id}`;
    dispatch(fetchCountryDetails({ url }));
  };
  const handleSearch = (e) => {
    setSearch(e.target.value);
  };
  const filteredHoliday = AllHolidays.filter((item) => (
    item.name.toLowerCase().includes(search.toLowerCase())
  ));

  return (
    <div>
      <div>
        <Navbar />
      </div>
      <div className="search-container">
        <p className="search-Title">Search For Country</p>
        <input className="search-Input-Area" placeholder="e.g. Andorra" onChange={handleSearch} />
      </div>
      {(AllHolidays.isLoading) ? <p>Loading...</p>

        : (
          <div className="individual-countries">
            {filteredHoliday.map((Holiday) => (
              <div key={Holiday.countryCode} className="one-country">
                <RightIcon />
                <div className="initials">{Holiday.countryCode}</div>
                <NavLink to="/CountryDetails" id={Holiday.countryCode} className="nav-to" onClick={handleDetailsFetch}>
                  {Holiday.name}
                  <br />
                  <span className="holidays-count">2 + Holidays</span>
                </NavLink>
                <div className="holiday-counter" />
              </div>
            ))}
          </div>
        )}
    </div>
  );
};

export default Countries;
