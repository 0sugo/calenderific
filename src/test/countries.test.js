import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom'; // Import the BrowserRouter component
import Countries from '../components/Countries';

const mockStore = configureStore([]);

test('renders the Countries component correctly', () => {
  const initialState = {
    allHolidays: {
      AllHolidays: [
        {
          countryCode: 'US',
          name: 'United States',
        },
        {
          countryCode: 'CA',
          name: 'Canada',
        },
      ],
      isLoading: false,
    },
  };

  const store = mockStore(initialState);

  const { container } = render(
    <Provider store={store}>
      <Router> {/* Wrap the component with the Router */}
        <Countries />
      </Router>
    </Provider>
  );

  expect(container).toMatchSnapshot();
});

test('renders the loading state correctly', () => {
  const initialState = {
    allHolidays: {
      AllHolidays: [],
      isLoading: true,
    },
  };

  const store = mockStore(initialState);

  const { container } = render(
    <Provider store={store}>
      <Router> {/* Wrap the component with the Router */}
        <Countries />
      </Router>
    </Provider>
  );

  expect(container).toMatchSnapshot();
});
