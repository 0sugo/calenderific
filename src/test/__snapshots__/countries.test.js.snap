// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Countries component correctly 1`] = `
<div>
   
  <div>
    <div>
      <div
        class="nav-container"
      >
        <a
          aria-current="page"
          class="back active"
          href="/"
        >
          <svg
            class="w-6 h-6 leftchev"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.75 19.5L8.25 12l7.5-7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </a>
        <h1
          class="topper"
        >
          World Holidays
        </h1>
      </div>
      <div
        class="hero-section"
      >
        <img
          alt="world"
          class="header-photo"
          src="globalphoto.png"
        />
        <div
          class="All-holidays"
        >
          <h3>
            <span
              class="before-text"
            >
              Sampled
            </span>
            <br />
             
            <span
              class="main-text"
            >
              2
            </span>
             
            <span
              class="after-text"
            >
              Countries
            </span>
          </h3>
        </div>
      </div>
    </div>
    <div
      class="search-container"
    >
      <p
        class="search-Title"
      >
        Search For Country
      </p>
      <input
        class="search-Input-Area"
        placeholder="e.g. Andorra"
      />
    </div>
    <div
      class="individual-countries"
    >
      <a
        class="one-country"
        data-testid="filtered-holiday"
        href="/CountryDetails"
        id="US"
      >
        <svg
          class="w-6 h-6 righter"
          fill="none"
          stroke="currentColor"
          stroke-width="1.2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.75 15l3-3m0 0l-3-3m3 3h-7.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <div
          class="initials"
        >
          US
        </div>
        <p
          class="nav-to"
        >
          United States
          <br />
          <span
            class="holidays-count"
          >
            2 + Holidays
          </span>
        </p>
        <div
          class="holiday-counter"
        />
      </a>
      <a
        class="one-country"
        data-testid="filtered-holiday"
        href="/CountryDetails"
        id="CA"
      >
        <svg
          class="w-6 h-6 righter"
          fill="none"
          stroke="currentColor"
          stroke-width="1.2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.75 15l3-3m0 0l-3-3m3 3h-7.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <div
          class="initials"
        >
          CA
        </div>
        <p
          class="nav-to"
        >
          Canada
          <br />
          <span
            class="holidays-count"
          >
            2 + Holidays
          </span>
        </p>
        <div
          class="holiday-counter"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`renders the loading state correctly 1`] = `
<div>
   
  <div>
    <div>
      <div
        class="nav-container"
      >
        <a
          aria-current="page"
          class="back active"
          href="/"
        >
          <svg
            class="w-6 h-6 leftchev"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.75 19.5L8.25 12l7.5-7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </a>
        <h1
          class="topper"
        >
          World Holidays
        </h1>
      </div>
      <div
        class="hero-section"
      >
        <img
          alt="world"
          class="header-photo"
          src="globalphoto.png"
        />
        <div
          class="All-holidays"
        >
          <h3>
            <span
              class="before-text"
            >
              Sampled
            </span>
            <br />
             
            <span
              class="main-text"
            >
              0
            </span>
             
            <span
              class="after-text"
            >
              Countries
            </span>
          </h3>
        </div>
      </div>
    </div>
    <div
      class="search-container"
    >
      <p
        class="search-Title"
      >
        Search For Country
      </p>
      <input
        class="search-Input-Area"
        placeholder="e.g. Andorra"
      />
    </div>
    <div
      class="individual-countries"
    />
  </div>
</div>
`;
